bindings/python/src/lib.rs

```diff
@@ -717,40 +727,60 @@ fn create_tensor(
     array: PyObject,
     device: &Device,
 ) -> PyResult<PyObject> {
-    match framework {
-        Framework::Pytorch | Framework::Numpy => Python::with_gil(|py| -> PyResult<PyObject> {
-            let module: &PyModule = match framework {
-                Framework::Numpy => NUMPY_MODULE.get(py),
-                Framework::Pytorch => TORCH_MODULE.get(py),
-                _ => unreachable!(),
+    Python::with_gil(|py| -> PyResult<PyObject> {
+        let module: &PyModule = match framework {
+            Framework::Pytorch => TORCH_MODULE.get(py),
+            _ => NUMPY_MODULE.get(py),
+        }
+        .ok_or_else(|| {
+            exceptions::PyException::new_err(format!("Could not find module {framework:?}",))
+        })?
+        .as_ref(py);
+        let frombuffer = module.getattr(intern!(py, "frombuffer"))?;
+        let dtype: PyObject = get_pydtype(module, dtype)?;
+        let kwargs = [
+            (intern!(py, "buffer"), array),
+            (intern!(py, "dtype"), dtype),
+        ]
+        .into_py_dict(py);
+        let tensor = frombuffer.call((), Some(kwargs))?;
+        let shape = shape.to_vec();
+        let shape: PyObject = shape.into_py(py);
+        let mut tensor: &PyAny = tensor.getattr(intern!(py, "reshape"))?.call1((shape,))?;
+        if device != &Device::Cpu {
+            let device: PyObject = device.clone().into_py(py);
+            let kwargs = PyDict::new(py);
+            tensor = tensor
+                .getattr(intern!(py, "to"))?
+                .call((device,), Some(kwargs))?;
+        }
```

Comment:
This should probably be done only in `Framework` is `Framework::Pytorch`

Reply:
Yes ! I need to handle the other framework differently (not sure if this lib should handle it or not though).